Option to enforce probe download instead of building it

francesco-racciatti opened the pull request:
This add an option to modify the workflow of the sysdig-probe-loader script.

When the environment variable `SYSDIG_FORCE_DOWNLOAD_PROBE` is defined,
the script tries to download the appropriate kernel/bpf probe, instead of building it.

gnosek commented:
I like it, but I'm afraid the indentation is getting (even more) excessive. Can we introduce some helper functions and call them conditionally instead of the large inline `if` bodies?

francesco-racciatti commented:
Yep, absolutely!
tks

iurly commented:
@gnosek I was also wondering if should try and have an overall variable to define the strategy in some self-explanatory way, instead of adding yet another cryptical boolean one which makes things even harder to decipher.
For instance, something like `SYSDIG_PROBE_STRATEGY`, `SYSDIG_PROBE_SOURCE`, `SYSDIG_PROBE_BEHAVIOR`.
Example values would be `"ebpf:build,download"` or `"kmod:download,none"`.
Of course we would have to retain compatibility if already-known variables are set, and infer the strategy from there.
cc: @balinaveen @alblasco

gnosek commented:
I'm not sure. I mean, I like the idea but we'll end up with a parser for these values written in shell and if you put these words in a single sentence and rearrange them a bit, they're going to spell trouble ;)

francesco-racciatti commented:
@gnosek 
In the meantime I refactored the whole script.

francesco-racciatti commented:
@iurly @gnosek
I refactored the script.

gnosek commented on sysdig-probe-loader:
```diff
@@ -557,24 +592,8 @@ load_bpf_probe() {
-		echo "* Failure to find a BPF probe"
-		exit 1
-	fi
+	make_symlink_bfp_probe
```
s/bfp/bpf/ :) (here and above)

francesco-racciatti replied:
yes, I agree, I exaggerated a little ... 
I will pay more attention! :)

gnosek reviewed (commented):
Apart from the one typo (bfp -> bpf) it looks good :)

With a change this large (basically rewriting the whole thing) multiple individual commits are nice but I think we've gone a bit too far in this case :)

For one thing, there are commits that are reverted immediately, like the load -> mount -> load rewording, and also there's a lot of comment/log changes, which I guess for the most part could be left alone

alblasco commented on sysdig-probe-loader:
```diff
@@ -48,7 +58,6 @@
 # Where:
 #   DOWNLOAD_URL_HOST = "download.sysdig.com" by default
 #   DOWNLOAD_REPOSITORY = "stable" by defualt
```
typo: defualt

francesco-racciatti replied:
Thanks, fixed!

francesco-racciatti commented:
@alblasco @gnosek 
Thanks!  I fixed the typos.

francesco-racciatti committed:
refactor: add helper functions to reduce indentation level

francesco-racciatti committed:
feat: add option to enforce probe downloading

When defining the environment variable SYSDIG_FORCE_DOWNLOAD_PROBE,
the probe loader skips the building phase and tries to directly download the probe.

gnosek reviewed (approved):
✅
